p2p/upgrade.go

```diff
@@ -89,37 +89,22 @@ func Upgrade(h host.Host, opts ...Opt) (*Host, error) {
 		opt(fh)
 	}
 	cfg := fh.cfg
-	bootnode, err := isBootnode(h, cfg.Bootnodes)
+	bootnode, err := isBootnode(h, cfg.Discovery.Bootnodes)
 	if err != nil {
 		return nil, fmt.Errorf("check node as bootnode: %w", err)
 	}
-	if fh.PubSub, err = pubsub.New(fh.ctx, fh.logger, h, pubsub.Config{
-		Flood:          cfg.Flood,
-		IsBootnode:     bootnode,
-		MaxMessageSize: cfg.MaxMessageSize,
-	}); err != nil {
+	if fh.PubSub, err = pubsub.New(fh.ctx, fh.logger, h, cfg.Pubsub, bootnode); err != nil {
```

Comment:
I also thought about moving the `isBootnode` function to the `pubsub` pkg. I think that the reason why it was placed here is that it uses the `bootnodes []string` field of the config, which is not available from `pubsub.New()`. However, we don't want to explicitly pass the list of boot nodes to `pubsub.New`. The whole list is not required there - it just needs information if we are a boot node or not.

Regarding creating separate `New(...)` and `NewBootnode(...)` I'm not sure if it is worth it. There would be 3 functions instead of just one (private `new(...)` accepting the boolean and public New(...) and `NewBootstrap(...)`). IMO it's an overkill for this usecase.

I can propose something different. Another way to get rid of `bools` (which I hate too 😉 ) is to use enums. Consider:
```go
package p2p

type NodeType uint8
const (
  Bootnode NodeType = iota
  RegularNode 
)

---

package pubsub
func New(..., nodeType NodeType) { ... }
```

Reply:
Using multiple "constructors" like `New` and `NewBootnode` feels more idiomatic to me in go, but I would be OK with an enum for the NodeType as well.